zlib-rs/src/inflate/writer.rs

```diff
@@ -140,6 +163,25 @@ impl<'a> Writer<'a> {
 
     #[inline(always)]
     pub fn copy_match(&mut self, offset_from_end: usize, length: usize) {
+        self.copy_match_dispatch::<{ CpuFeatures::NONE }>(offset_from_end, length)
+    }
+
+    #[inline(always)]
+    pub fn copy_match_dispatch<const FEATURES: usize>(
+        &mut self,
+        offset_from_end: usize,
+        length: usize,
+    ) {
+        match FEATURES {
+            #[cfg(target_arch = "x86_64")]
+            CpuFeatures::AVX2 => {
+                self.copy_match_help::<core::arch::x86_64::__m256i>(offset_from_end, length)
+            }
+            _ => self.copy_match_fallback(offset_from_end, length),
+        }
+    }
+
+    pub fn copy_match_fallback(&mut self, offset_from_end: usize, length: usize) {
         // NOTE: the dynamic check for avx512 makes avx2 slower. Measure this carefully before re-enabling
         //
         //        #[cfg(target_arch = "x86_64")]
```

Comment:
The `is_enabled_avx2` branch can be removed from this function, right? If avx2 was supported, `copy_match_dispatch` would not call `copy_match_fallback`.

Reply:
not quite, we still use the `NONE` version in the logic of `len_and_friends`